crates/stackable-operator/src/utils/cluster_domain.rs

```diff
@@ -0,0 +1,183 @@
+use std::{env, path::Path, str::FromStr, sync::OnceLock};
+
+use snafu::{OptionExt, ResultExt, Snafu};
+use tracing::instrument;
+
+use crate::commons::networking::DomainName;
+
+const KUBERNETES_CLUSTER_DOMAIN_ENV: &str = "KUBERNETES_CLUSTER_DOMAIN";
+const KUBERNETES_SERVICE_HOST_ENV: &str = "KUBERNETES_SERVICE_HOST";
+
+const KUBERNETES_CLUSTER_DOMAIN_DEFAULT: &str = "cluster.local";
+const RESOLVE_CONF_FILE_PATH: &str = "/etc/resolv.conf";
+
+// TODO (@Techassi): Do we even need this many variants? Can we get rid of a bunch of variants and
+// fall back to defaults instead? Also trace the errors
+#[derive(Debug, Snafu)]
+pub enum Error {
+    #[snafu(display("failed to read resolv config from {RESOLVE_CONF_FILE_PATH}"))]
+    ReadResolvConfFile { source: std::io::Error },
+
+    #[snafu(display("failed to parse {cluster_domain:?} as domain name"))]
+    ParseDomainName {
```

Comment:
I would rename this to make it explicit (to the developer) that the _user supplied_ domain could not be parsed.

```suggestion
    ParseUserSuppliedDomainName {
```

Reply:
Decided not to do this, as we are emitting errors for both user input and searching via resolv.conf